program testBadInput5;
#include("stdlib.hhf")

static
    input:    int32;

begin testBadInput5;

    // This forever loop repeats until the user enters
    // a good integer and the BREAK statement below
    // exits the loop. Note that BREAK statment
    // appears in an UNPROTECTED section of the TRY..ENDTRY
    // statement.

    forever
        try
            stdout.put("Enter an integer value:");
            stdin.get(input);
            stdout.put("The first input value was:", input, nl);

        unprotected
            break;
        exception(ex.ValueOutOfRange)
            stdout.put("The value was too larget, reenter." nl);
        exception(ex.ConversionError)
            stdout.put("The input contained illegal characters, reenter." nl);
        endtry;
    endfor;

    // Note that the following code is outside the loop and there
    // is not TRY...ENDTRY statement protecting this code.
    stdout.put("Enter another number:");
    stdin.get(input);
    stdout.put("The new number is:", input, nl);
    
end testBadInput5;
    